src/antd/InputNumber.tsx

```diff
@@ -0,0 +1,84 @@
+import { ReactElement } from "react"
+import { ControlProps, RendererProps } from "@jsonforms/core"
+import { InputNumber as AntdInputNumber } from "antd"
+import { coerceToInteger, coerceToNumber, decimalToPercentage, percentageStringToDecimal } from "../controls/utils"
+
+type InputNumber = ReactElement<typeof AntdInputNumber>
+type AntdInputNumberProps = React.ComponentProps<typeof AntdInputNumber>
+type InputNumberProps = AntdInputNumberProps & RendererProps & ControlProps
+
+export const InputNumber = (props: InputNumberProps): InputNumber => {
+  const schema = props.schema
+  const ariaLabel = props.label || schema.description || "Value"
+
+  const defaultValue = schema.default as number | undefined
+  const isDataNonNullObject = typeof props.data === "object" && props.data !== undefined && props.data !== null
+  const isDataEmptyObj = isDataNonNullObject ? Object.keys(props.data as object).length === 0 : false
+  const value = props.data === undefined || isDataEmptyObj ? defaultValue : props.data as number
+
+  const numberType = schema.type
+  const isInteger = (typeof numberType === "string" && numberType === "integer") || (Array.isArray(numberType) && numberType.includes("integer"))
+  const handleChange = (value: number | null) => {
+    if (typeof value === "number") {
+      if (isInteger) {
+        props.handleChange(props.path, coerceToInteger(value))
+      } else {
+        props.handleChange(props.path, coerceToNumber(value))
+      }
+    } else {
+      props.handleChange(props.path, value)
+    }
+  }
+
+  const addonAfter = props.uischema.options?.addonAfter as string | undefined
```

Comment:
is this cast still necessary?

Reply:
Yes, it's considered an `any` until I do this case. It's essentially the same as what we decided to do [here](https://github.com/great-expectations/jsonforms-antd-renderers/blob/bda0b367aa53a0f3ecb0466b7be5a1bfe1f26074/src/controls/AlertControl.tsx#L6).

Reply:
I added a new type `InputNumberOptions` to `ui-schema.ts` to precisely follow the pattern established by `AlertControl`. There is still a cast here though, the cast is just on `options` now and it passes through the props types.

Reply:
ah, it's a lot easier to read if you pull off an object and type it:
```
const options: Maybe<InputNumberOptions> = props.uischema.options as InputNumberOptions
const addonAfter = options?.addonAfter
```

Reply:
I'm sorry, how is this different than what I did?

Reply:
typos like ```props.uischema.options?.adonafter as string | undefined``` doesn't get caught but it would in my example

Reply:
Ok I don't think you've seen the updated version. This is now outdated.